pcdsdevices/lasers/rfof.py

```diff
@@ -13,17 +13,242 @@
+    las_pwr_err = Cpt(EpicsSignalRO, ':LAS_PWR_ERR', kind='normal',
+                      doc='Laser is switched off')
+    phase_err = Cpt(EpicsSignalRO, ':PHASE_ERR', kind='normal',
+                    doc='Phase loops are not locked')
+    fan_spd_err = Cpt(EpicsSignalRO, ':FAN_SPD_ERR', kind='normal',
+                      doc='Fan revolutions are too low')
+    pwr_supply_i_err = Cpt(EpicsSignalRO, ':PWR_SUPPLY_I_ERR', kind='normal',
+                           doc='Power supply currents are too high')
+    tx_eth_err = Cpt(EpicsSignalRO, ':TX_ETH_SYNC_ERR', kind='normal',
+                     doc='Tx: Ethernet sync between units \
+                          lost for more than 10 min')
+    rx_eth_err = Cpt(EpicsSignalRO, ':RX_ETH_SYNC_ERR', kind='normal',
+                     doc='Rx: Ethernet sync between units \
+                          lost for more than 10 min')
+    tx_input_err = Cpt(EpicsSignalRO, ':TX_RF_INPUT_ERR', kind='normal',
+                       doc='RF input is out of range for more than 60s')
+    rx_output_err = Cpt(EpicsSignalRO, ':RX_RF_OUTPUT_ERR', kind='normal',
+                        doc='RF output is out of range for more than 10 min')
+    tx_temp_sens_err = Cpt(EpicsSignalRO, ':TX_TEMP_SENS_ERR', kind='normal',
+                           doc='Tx: Temperature sensors are not working properly; \
+                                TEC protection activated')
+    rx_temp_sens_err = Cpt(EpicsSignalRO, ':RX_TEMP_SENS_ERR', kind='normal',
+                           doc='Rx: Temperature sensors are not working properly; \
+                                TEC protection activated')
+    tx_rf_idn_err = Cpt(EpicsSignalRO, ':TX_RF_IDN_ERR', kind='normal',
+                        doc='Tx: RF control identification failed')
+    tx_rf_sens_err = Cpt(EpicsSignalRO, ':TX_RF_SENS_ERR', kind='normal',
+                         doc='Tx: RF control sensor failure')
+    tx_arm_idn_err = Cpt(EpicsSignalRO, ':TX_ARM_IDN_ERR', kind='normal',
+                         doc='Tx: Arm control identification failed')
+    tx_arm_sens_err = Cpt(EpicsSignalRO, ':TX_ARM_SENS_ERR', kind='normal',
+                          doc='Tx: Arm control sensor failure')
+    unlock_err = Cpt(EpicsSignalRO, ':UNLOCK_ERR', kind='normal',
+                     doc='System is unlocked for more than 15 min')
+    rx_rf_idn_err = Cpt(EpicsSignalRO, ':RX_RF_IDN_ERR', kind='normal',
+                        doc='Rx: RF control identification failed')
+    rx_rf_sens_err = Cpt(EpicsSignalRO, ':RX_RF_SENS_ERR', kind='normal',
+                         doc='Rx: RF control sensor failure')
+    rx_arm_idn_err = Cpt(EpicsSignalRO, ':RX_ARM_IDN_ERR', kind='normal',
+                         doc='Rx: Arm control identification failed')
+    rx_arm_sens_err = Cpt(EpicsSignalRO, ':RX_ARM_SENS_ERR', kind='normal',
+                          doc='Rx: Arm control sensor failure')
+
+
+class ItechRfofBase(ItechRfofTx, ItechRfofRx,
```

Comment:
This one still needs addressing. Consider this example as to what inheritance does for you:

```python
class MyBaseA:
    a = "this is A"


class MyBaseB:
    b = "this is B"


class MyClass(MyBaseA, MyBaseB):
    """
    MyClass inherits from (or "subclasses") both MyBaseA and MyBaseB. 

    This means all of the elements from MyBaseA and MyBaseB will be included in
    MyClass.

    This is in addition to whatever I put in the class body below.
    """
    c = "this is from MyClass"


print(MyClass().a)   # -> "this is A"
print(MyClass().b)   # -> "this is B"
print(MyClass().c)   # -> "this is from MyClass"
```

With the above in mind, now recall what ophyd does with its "sub-devices" as components:

```python
from ophyd import EpicsSignal, Device, Component as Cpt


class SubDeviceA(Device):
    a_value = Cpt(EpicsSignal, ":suffix")

class SubDeviceB(Device):
    b_value = Cpt(EpicsSignal, ":suffix")

class MyDevice(Device):
    cpt_a = Cpt(SubDeviceA, ":prefix:for:a")
    cpt_b = Cpt(SubDeviceB, ":prefix:for:b")


device = MyDevice("base:prefix", name="device")
print(device.cpt_a.a_value.pvname)   # ->  base:prefix:prefix:for:a:suffix
print(device.cpt_b.b_value.pvname)   # -> base:prefix:prefix:for:b:suffix
print(device.component_names)   # -> ('cpt_a', 'cpt_b')
# device.a_value does not exist (and it shouldn't!) - it's part of the subdevice.
# device.b_value does not exist (and it shouldn't!) - it's part of the subdevice.
```

OK, now what happens if we combine the subclassing _with_ using SubDevice?

```python
class MyDevice(SubDeviceA, SubDeviceB):  # <-- we subclassed
    cpt_a = Cpt(SubDeviceA, "prefix:for:a:")  # <-- and included components
    cpt_b = Cpt(SubDeviceB, "prefix:for:b:")


device = MyDevice("base:prefix:", name="device")
print(device.cpt_a.a_value.pvname)   # ->  base:prefix:prefix:for:a:suffix
print(device.a_value.pvname)  # -> base:prefix::suffix  (!!)
```

Wait, ``a_value`` is on ``device`` AND ``device.cpt_a``, too? This is not good! Look, the prefixes differ confusingly, too...

```python
print(device.component_names)  # -> ('b_value', 'a_value', 'cpt_a', 'cpt_b')
```

These should only be in one spot - and the correct spot would be to keep them as sub-device components (in my opinion) as you could then access components by way of `device.transmitter.tx_*`.

I hope the above makes it a bit clearer why you shouldn't be doing both. Feel free to ask follow-up questions if not!